BlockstackInstance+ObjC.swift

```diff
@@ -0,0 +1,107 @@
+
+@objc public class ProfileObject: NSObject {
+    
+    public let type: String?
+    public let context: String?
+    public let name: String?
+    public let profileDescription: String?
+    
+    public init(profile: Profile) {
+        type = profile.type
+        context = profile.context
+        name = profile.name
+        profileDescription = profile.description
+    }
+    
+}
+
+@objc public class UserDataObject: NSObject {
+    
+    public let jti: String?
+    public let iat, exp: Int?
+    public let iss: String?
+    public var privateKey: String?
+    public let publicKeys: [String]?
+    public let username, email, coreToken, profileURL, hubURL, version: String?
+    public let claim: ProfileObject?
+    public var profile: ProfileObject?
+    
+    public init(userData: UserData) {
+        jti = userData.jti
+        iat = userData.iat
+        exp = userData.exp
+        iss = userData.iss
+        privateKey = userData.privateKey
+        publicKeys = userData.publicKeys
+        username = userData.username
+        email = userData.email
+        coreToken = userData.coreToken
+        profileURL = userData.profileURL
+        hubURL = userData.hubURL
+        version = userData.version
+        
+        if let claimStruct = userData.claim {
+            claim = ProfileObject(profile: claimStruct)
+        } else {
+            claim = nil
+        }
+        
+        if let profileStruct = userData.profile {
+            profile = ProfileObject(profile: profileStruct)
+        } else {
+            profile = nil
+        }
+    }
+    
+}
+
+public extension BlockstackInstance {
+    
+    @objc func signIn(redirectURI: String,
+                      appDomain: URL,
+                      manifestURI: URL? = nil,
+                      scopes: Array<String>,
+                      completion: ((UserDataObject?, Error?, Bool) -> Void)?) {
+        
+        let translatedCompletion: (AuthResult) -> Void = { authResult in
+            switch authResult {
+            case .success(let userData): completion?(UserDataObject(userData: userData), nil, false)
+            case .failed(let error): completion?(nil, error, false)
+            case .cancelled: completion?(nil, nil, true)
+            }
+        }
+        signIn(redirectURI: redirectURI, appDomain: appDomain, manifestURI: manifestURI, scopes: scopes, completion: translatedCompletion)
+    }
+    
+    @objc func signIn(redirectURI: String,
+                      appDomain: URL,
+                      manifestURI: URL? = nil,
+                      completion: ((UserDataObject?, Error?, Bool) -> Void)?) {
+        signIn(redirectURI: redirectURI, appDomain: appDomain, manifestURI: manifestURI, scopes: ["store_write"], completion: completion)
+    }
+    
+    @objc public func putFile(path: String, content: Dictionary<String, String>, completion: @escaping (String?, Error?) -> Void) {
+        putFile(path: path, content: content, completion: completion)
```

Comment:
This ends up recursing and calling itself. I changed the name on line 91 but there's probably a more clever solution using the @objc tag and it's options.  Same problem on line 96 below too (getFile).

Reply:
Good that you spotted it @AC-FTW, thanks! I forgot to push that fix 🙄 😓 

Fixed it in: 

https://github.com/blockstack/blockstack-ios/pull/10/commits/c36c90b16f7c266158d053cd500174b6739c391b